### PR TITLE
Cleanup and modernize Browser sample

### LINKED ISSUE #212: Sample Browser is not user friendly
Here is more user friendly version.

[Browser.zip](https://github.com/jmdns/jmdns/files/5519071/Browser.zip)

### PR DESCRIPTION
**Pull request by melloware**
Fix #212: Cleanup and modernize Browser sample

image

### CODE REVIEW

File: src/sample/java/com/strangeberry/jmdns/tools/Browser.java [hunk 1]
```diff
@@ -43,42 +48,40 @@
...
-    JmmDNS                    jmmdns;
-    // Vector headers;
-    String                    type;
-    DefaultListModel          types;
-    JList                     typeList;
-    DefaultListModel          services;
-    JList                     serviceList;
-    JTextArea                 info;
+    static JmmDNS jmmdns;
+    static MdnsBrowser instance;
```
Comment (janossch) [inline src/sample/java/com/strangeberry/jmdns/tools/Browser.java:61]: I think this should be `Browser`

```suggestion
    static Browser instance;
```

File: src/sample/java/com/strangeberry/jmdns/tools/Browser.java [hunk 2]
```diff
@@ -108,157 +111,149 @@ public class Browser extends JFrame implements ServiceListener, ServiceTypeListe
...
+    /**
+     * Called when a service is added.
+     *
+     * @param event The ServiceEvent containing information about the added service
      */
     @Override
     public void serviceAdded(ServiceEvent event) {
         final String name = event.getName();
-
-        System.out.println("ADD: " + name);
```
Comment (janossch) [inline src/sample/java/com/strangeberry/jmdns/tools/Browser.java:144]: Maybe it is just my personal preference to like reading logs but what I suggest is to create a private method where you do the logging. By default it is switched off and if someone who feels the irresistible need could switch it back on.

File: src/sample/java/com/strangeberry/jmdns/tools/Browser.java [hunk 3]
```diff
@@ -21,9 +21,14 @@
 import java.awt.BorderLayout;
 import java.awt.Color;
 import java.awt.Container;
+import java.awt.Dimension;
+import java.awt.Font;
 import java.awt.GridLayout;
+import java.awt.Toolkit;
 import java.io.IOException;
+import java.io.Serial;
```
Comment (janossch) [inline src/sample/java/com/strangeberry/jmdns/tools/Browser.java:29]: `java.io.Serial` is this really used?
```suggestion
```

**Review by janossch — CHANGES_REQUESTED** [review]

**Review by janossch — APPROVED** [review]
Looks good to me!

File: src/sample/java/samples/Browser.java [hunk 4]
```diff
@@ -0,0 +1,311 @@
+package javax.jmdns.tools;
```
Comment (kaikreuzer) [inline src/sample/java/samples/Browser.java:1]: But now there is no license header at all...?

**Comment by kaikreuzer** [comment]
It would be nice if you could keep the code history in place by doing two separate commits: One merely moving the file to the new location (git can than track it and associate the old and new location) and one with the code changes.

**Comment by melloware** [comment]
Yep fixing the headers now

**Comment by melloware** [comment]
Weird i moved them all the same in IntelliJ its funny two were moves and one was an Add and Delete?

**Comment by melloware** [comment]
OK from what I understand this is Git default behavior

**Pull request closed without merging**